qemu/tests/numa_stress.py

```diff
@@ -134,23 +189,23 @@ def run(test, params, env):
                 test.fail("Can not allocate memory in node %s."
                           " Error message:%s" % (numa_node_malloc,
                                                  str(error_msg)))
-        error_context.context("Run memory heavy stress in guest", logging.info)
-        stress_test = utils_test.VMStress(vm, "stress", params, stress_args=stress_args)
-        stress_test.load_stress_tool()
-        error_context.context("Get the qemu process memory use status",
-                              logging.info)
-        node_after, memory_after = max_mem_map_node(host_numa_node, qemu_pid)
-        if node_after == most_used_node and memory_after >= memory_used:
-            test.fail("Memory still stick in node %s" % numa_node_malloc)
         else:
-            most_used_node = node_after
-            memory_used = memory_after
-        stress_test.unload_stress()
-        stress_test.clean()
-        utils_misc.umount("none", tmpfs_path, "tmpfs")
-        funcatexit.unregister(env, params.get("type"), utils_misc.umount,
-                              "none", tmpfs_path, "tmpfs")
-        session.cmd("sync; echo 3 > /proc/sys/vm/drop_caches")
-        utils_memory.drop_caches()
+            error_context.context("Run memory heavy stress in guest",
```

Comment:
After move this to the `else`, stress test will not be executed when `No space` with the `numa_node_malloc`.
This is not what we want if i understand correctly.

Reply:
Yes, you are right, though after update tmpfs_size, this may not happen. Do you have any suggestion? I moved the code to `else` to make the umount in `finally` block.

Reply:
Yes, the tmpfs_size it the free mem of the most used node.  Considering the critical condition, to be more safe, better keep it.
For the second issue, how do you think about involve a new `try...finally` block ?

Reply:
If add a new try block, the code would be like:
'''
try:
    try:
        utils_misc.mount()
    except Exception as error_msg:
        ...
    run_stress
finally:
    utils_misc.umount()
'''
Do you think it's acceptable?

Reply:
It is acceptable for me, a normal usage i think, but a little different with above:
```
try:
    utils_misc.mount()
    funcatexit.register()
    try:
        process.system()
    except Exception as e:    
        ......
    run_stress
finally:
    utils_misc.umount()
```

Reply:
Thanks, that makes more sense. Updated the exception as well.